src/storage-client/src/storage_collections.rs

```diff
@@ -2451,10 +2453,15 @@ async fn finalize_shards_task<T>(
         finalized_shards,
         persist_location,
         persist,
+        read_only,
     }: FinalizeShardsTaskConfig,
 ) where
     T: TimelyTimestamp + Lattice + Codec64,
 {
+    if read_only {
+        return;
```

Comment:
Mind throwing an `info!("disabling shard finalization in read only mode")` in here?

Reply:
Done